src/core/util.js

```diff
@@ -147,4 +149,18 @@ util.endsWith = function (str, suffix) {
   return str.indexOf(suffix, str.length - suffix.length) !== -1;
 };
 
+util.checkRequiredOpts = function (actualOpts, requiredOpts, from) {
+  _.each(requiredOpts, function (item) {
+    if (_.isUndefined(actualOpts[item])) {
+      throw new Error(item + ' is required' + (from ? ' to initialize ' + from : ''));
+    }
+  });
+};
+
+util.setAsPrivateProperties = function (opts, context) {
+  _.each(_.keys(opts), function (item) {
+    context['_' + item] = opts[item];
+  });
+};
+
```

Comment:
^^^
I've divided `checkAndBuildOptions` in two functions: one to check that all required options are present and another one to set them as private properties.

I've moved them to `util` as well. You can check how they're used at `dataviews-factory`, for instance.